File: Python/1238. Circular Permutation in Binary Representation.py
class Solution:
    def circularPermutation(self, n: int, start: int) -> List[int]:
        def gray():
            for i in range(2 ** n): yield i ^ (i >> 1)
        
        temp = list(gray())
        ptr = temp.index(start)

        return temp[ptr:] + temp[:ptr]

File: Java/1238. Circular Permutation in Binary Representation.java
class Solution {
    public List<Integer> circularPermutation(int n, int start) {
        List<Integer> ans = new ArrayList<>();

        for(int i = 0; i < (1 << n); i++) ans.add(i ^ (i >> 1));
        while((int)ans.get(0) != start) ans.add(ans.remove(0));

        return ans;
    }
}
